app/v1/controllers/CreateAmendUkSavingsAnnualSummaryController.scala

```diff
@@ -49,27 +49,24 @@ class CreateAmendUkSavingsAnnualSummaryController @Inject() (val authService: En
     authorisedAction(nino).async(parse.json) { implicit request =>
       implicit val ctx: RequestContext = RequestContext.from(idGenerator, endpointLogContext)
 
-      val rawData = CreateAmendUkSavingsAnnualSummaryRawData(
-        nino = nino,
-        taxYear = taxYear,
-        savingsAccountId = savingsAccountId,
-        body = AnyContentAsJson(request.body)
-      )
+      val validator = validatorFactory.validator(nino, taxYear, savingsAccountId, request.body)
 
       val requestHandler = RequestHandler
-        .withParser(parser)
-        .withService(service.createAmend)
-        .withNoContentResult(OK)
-        .withAuditing(AuditHandler.flattenedAuditing(
+        .withValidator(validator)
+        .withService { req =>
+          service.createAmend(req)
+        }
+        .withAuditing(AuditHandler(
           auditService = auditService,
-          auditType = "createAmendUkSavingsAnnualSummary",
-          transactionName = "create-and-amend-uk-savings-account-annual-summary",
-          params = Map("versionNumber" -> "1.0", "nino" -> nino, "taxYear" -> taxYear, "savingsAccountId" -> savingsAccountId),
+          auditType = "CreateAmendUkSavingsAnnualSummary",
+          transactionName = "create-amend-uk-savings-annual-summary",
+          params = Map("nino" -> nino, "taxYear" -> taxYear),
           requestBody = Some(request.body),
-          includeResponse = true
+          includeResponse = true,
+          apiVersion = Version1
```

Comment:
should be:    apiVersion = Version(request)

(as the controller can't really assume its version, might one day be coming to this same controller from a v2.routes etc)